public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -236,3 +237,52 @@ export function getLogQueryFromMetricsQuery(query: string): string {
 
   return selector + pipelineExpr;
 }
+
+export function isQueryWithLabelFilter(query: string): string {
+  const tree = parser.parse(query);
+  let labelName = '';
+
+  tree.iterate({
+    enter: ({ type, node }): false | void => {
+      if (type.id === LabelFilter) {
+        const label = node.getChild(Matcher)?.getChild(Identifier);
```

Comment:
I think once we get here, we should immediately return true. Otherwise, we get stuck here:

![suggested twice](https://user-images.githubusercontent.com/1069378/208645861-edac8119-088d-4fd8-8282-02a0a5ee88f9.png)

Reply:
Also, this function is called `isQueryWithLabelFilter` but is returning a string. `isX` must return a boolean.

Reply:
> Also, this function is called isQueryWithLabelFilter but is returning a string. isX must return a boolean.

@matyax good catch, its changed now